policy-controller/k8s/status/src/index.rs

```diff
@@ -0,0 +1,246 @@
+use crate::{
+    http_route::{ParentReference, RouteBinding},
+    resource_id::ResourceId,
+};
+use ahash::{AHashMap as HashMap, AHashSet as HashSet};
+use chrono::offset::Utc;
+use linkerd_policy_controller_k8s_api::{self as k8s, gateway, ResourceExt};
+use parking_lot::RwLock;
+use std::{collections::hash_map::Entry, sync::Arc};
+use tokio::sync::mpsc::{UnboundedReceiver, UnboundedSender};
+
+const POLICY_API_GROUP: &str = "policy.linkerd.io";
+const POLICY_API_VERSION: &str = "policy.linkerd.io/v1alpha1";
+const STATUS_CONTROLLER_NAME: &str = "policy.linkerd.io/status-controller";
+
+pub type SharedIndex = Arc<RwLock<Index>>;
+
+pub struct Controller {
+    client: k8s::Client,
+    index: SharedIndex,
+    updates: UnboundedReceiver<Update>,
+}
+
+pub struct Index {
+    updates: UnboundedSender<Update>,
+
+    http_routes: HashMap<ResourceId, RouteBinding>,
+    servers: HashSet<ResourceId>,
+}
+
+pub enum Update {
+    HttpRoute(ResourceId),
+    Server,
+}
+
+impl Controller {
+    pub fn new(
+        client: k8s::Client,
+        index: SharedIndex,
+        updates: UnboundedReceiver<Update>,
+    ) -> Self {
+        Self {
+            client,
+            index,
+            updates,
+        }
+    }
+
+    pub async fn process_updates(mut self) {
```

Comment:
while watching your recorded talk it seemed like we might need a comment here

// TODO: do we requeue if an update fails?

Reply:
658f36223